backend/main.go

```diff
@@ -1,215 +1,241 @@
-		WriteTimeout: 15 * time.Second,
-		ReadTimeout:  15 * time.Second,
-	}
-
-	l.Fatal(srv.ListenAndServe())
+    // Import filters
+    filterRaw, err := os.ReadFile("filter.txt")
+    if err != nil {
+        log.Println("could not import filter.txt; NSFW entries will not be filtered")
+    } else {
+        log.Println("imported filter.txt")
+    }
+    
+    filter = strings.Split(string(filterRaw), "\r\n")
+    
+    // Connect Flashpoint database
+    flashpointDB, errorFlashpointDB = sql.Open("sqlite3", "flashpoint.sqlite")
+    if errorFlashpointDB != nil {
+        log.Fatal(errorFlashpointDB)
+    }
+    
+    defer flashpointDB.Close()
+    log.Println("connected to flashpoint.sqlite")
+    
+    // Create vote database if it doesn't exist, then connect
+    if _, err := os.Stat("votes.sqlite"); errors.Is(err, os.ErrNotExist) {
+        os.Create("votes.sqlite")
+        log.Println("created votes.sqlite")
+    }
+    
+    votesDB, errorVotesDB = sql.Open("sqlite3", "votes.sqlite")
+    if errorVotesDB != nil {
+        log.Fatal(errorVotesDB)
+    }
+    
+    defer votesDB.Close()
+    log.Println("connected to votes.sqlite")
+    
+    // Create vote table if it doesn't exist
+    if _, err := votesDB.Exec(`
+        CREATE TABLE IF NOT EXISTS votes (
+            id      VARCHAR(36) PRIMARY KEY,
+            working INTEGER,
+            broken  INTEGER
+        )
+    `); err != nil {
+        log.Fatal(err)
+    }
+        
+    // Set up and start server
+    http.HandleFunc("/random",   randomHandler )
+    http.HandleFunc("/get/",     getHandler    )
+    http.HandleFunc("/working/", workingHandler)
+    http.HandleFunc("/broken/",  brokenHandler )
+    
+    server := &http.Server{
+        Addr: "127.0.0.1:8985",
+        WriteTimeout: 15 * time.Second,
+        ReadTimeout:  15 * time.Second,
+    }
+    
+    log.Printf("server started at %v\n", server.Addr)
+    log.Fatal(server.ListenAndServe())
 }
 
-func workyHandler(w http.ResponseWriter, r *http.Request) {
-	vars := mux.Vars(r)
-	uuid := vars["uuid"]
-
-	if err := addComment(uuid, "works", ""); err != nil {
-		l.Error(err)
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
-
-	w.WriteHeader(http.StatusOK)
+// Return JSON-formatted info about a random Flashpoint entry
+func randomHandler(w http.ResponseWriter, r *http.Request) {
+    var entry Entry
+    
+    // If the NSFW filter is active, "re-roll" until a non-NSFW entry is picked
+    ParentLoop:
+        for {
+            var tags string
+            
+            fpRow := flashpointDB.QueryRow(`
+                SELECT   id, title, launchCommand, tagsStr 
+                FROM     game 
+                WHERE    launchCommand LIKE '%.swf'
+                ORDER BY random()
+                LIMIT    1
+            `)
+            fpErr := fpRow.Scan(&entry.UUID, &entry.Title, &entry.LaunchCommand, &tags)
+            if fpErr != sql.ErrNoRows && fpErr != nil {
+                log.Fatal(fpErr)
```

Comment:
not a fan at all, this will just dump panics into the log and return 500, but I accept it for now